src/solvers/flattening/arrays.cpp

```diff
@@ -209,24 +211,23 @@ void arrayst::collect_arrays(const exprt &a)
   }
   else if(a.id()==ID_typecast)
   {
-    // cast between array types?
-    DATA_INVARIANT(
-      a.operands().size()==1,
-      "typecast must have one operand");
+    const auto &typecast_op = to_typecast_expr(a);
 
+    // cast between array types?
     DATA_INVARIANT(
-      a.op0().type().id()==ID_array,
-      ("unexpected array type cast from "+
-       a.op0().type().id_string()).c_str());
+      typecast_op.type().id() == ID_array,
```

Comment:
Looks like you really did want `typecast(...).op()` above, this is a change of behaviour

Reply:
indeed -- fixed